middle_end/flambda2/to_cmm/to_cmm_set_of_closures.ml

```diff
@@ -224,40 +230,85 @@ end = struct
           in
           List.rev_append (P.define_symbol (R.symbol res function_symbol)) acc
       in
-      (* We build here the **reverse** list of fields for the function slot *)
-      match closure_code_pointers with
-      | Full_application_only ->
-        if size <> 2
-        then
-          Misc.fatal_errorf
-            "fill_slot: Function slot %a is of size %d, but it is used to \
-             store code ID %a which is classified as Full_application_only (so \
-             the expected size is 2)"
-            Function_slot.print function_slot size Code_id.print code_id;
-        let acc =
-          P.int ~dbg closure_info :: P.term_of_symbol ~dbg code_symbol :: acc
+      match code_id with
+      | Code_id code_id -> (
+        let code_symbol =
+          R.symbol_of_code_id res ~currently_in_inlined_body:false code_id
         in
-        ( acc,
-          Backend_var.Set.empty,
-          slot_offset + size,
-          env,
-          res,
-          Ece.pure,
-          updates )
-      | Full_and_partial_application ->
-        if size <> 3
+        let (kind, params_ty, result_ty), closure_code_pointers, dbg =
+          get_func_decl_params_arity env code_id
+        in
+        let closure_info =
+          C.closure_info' ~arity:(kind, params_ty)
+            ~startenv:(startenv - slot_offset) ~is_last:last_function_slot
+        in
+        (* We build here the **reverse** list of fields for the function slot *)
+        match closure_code_pointers with
+        | Full_application_only ->
+          if size < 2
```

Comment:
I don't understand this change: particularly, why could it happen that `size = 3` ? and why it is correct for such function slots to exist ?

Reply:
This was a consequence of a previous version when deleted slot offsets had no size; I've removed it.